packages/Webkul/Rule/src/Helpers/Validator.php

```diff
@@ -251,4 +251,28 @@ public function validateAttribute($condition, $attributeValue)
 
         return $result;
     }
+
+    /**
+     * Validate the condition value against a multi dimensional array recursively
+     *
+     * @param array  $attributeValue
+     * @param string $conditionValue
+     *
+     * @return bool
+     */
+    private static function validateArrayValues(array $attributeValue, string $conditionValue): bool
+    {
+        $result = in_array($conditionValue, $attributeValue, true);
+        if ($result === false) {
+            foreach ($attributeValue as $subValue) {
+                if (is_array($subValue)) {
+                    $result = self::validateArrayValues($subValue, $conditionValue);
+                    if ($result === true) {
+                        break;
+                    }
+                }
+            }
+        }
+        return $result;
```

Comment:
I think early returns should be used:

```suggestion
        if (in_array($conditionValue, $attributeValue, true) === true) {
            return true;
        }
        
        foreach ($attributeValue as $subValue) {
            if (is_array($subValue)) {
                if (self::validateArrayValues($subValue, $conditionValue) === true) {
                    return true;
                }
            }
        }
        return false;
```

Reply:
Alright, always thinking positively :D Thanks!